widget/view.controller.js

```diff
@@ -85,208 +70,146 @@
-          renderNoRecordMessage();
+            links.push(linkData);
+          }
         }
-      }, function (error) {
-        $scope.processing = false;
-        renderNoRecordMessage();
-      }).finally(function () {
-        $scope.processing = false;
       });
+
+      if(nodes.length > 0 && links.length>0){
+        chartData.nodes = nodes;
+        chartData.links = links;
+        renderSankeyChart();
+      }else{
+        errorMessage = $scope.viewWidgetVars.MESSAGE_NO_NODES_LINKS;
+        renderNoRecordMessage('Nodes and Links not created by the given data')
+      }
+    
     }
 
-    function getResourceAggregate() {
-      var defer = $q.defer();
-      var dataFilters = getFilters();
-      var queryObject = {
-        sort: [{
-          field: 'total',
-          direction: 'DESC'
-        }],
-        aggregates: [
-          {
-            'operator': 'countdistinct',
-            'field': '*',
-            'alias': 'total'
-          },
-          {
-            'operator': 'groupby',
-            'alias': 'source',
-            'field': $scope.config.sourceNodesField
-          },
-          {
-            'operator': 'groupby',
-            'alias': 'target',
-            'field': $scope.config.targetNodesField + '.itemValue'
-          },
-          {
-            'operator': 'groupby',
-            'alias': 'relation',
-            'field': $scope.config.relation + '.' + $scope.config.relationshipNodesField + '.itemValue'
-          },
-          {
-            'operator': 'groupby',
-            'alias': 'targetColor',
-            'field': $scope.config.targetNodesField + '.color'
-          },
-          {
-            'operator': 'groupby',
-            'alias': 'relationColor',
-            'field': $scope.config.relation + '.' + $scope.config.relationshipNodesField + '.color'
-          }
-        ],
-        relationship: true,
-        filters: [dataFilters]
+    //to populate data for custom module
+    function populateCustomData() {
+      var filters = {
+        query: $scope.config.query
       };
-      var _queryObj = new Query(queryObject);
-      $http.post(API.QUERY + $scope.config.resource, _queryObj.getQuery(true)).then(function (response) {
-        defer.resolve(response.data);
-      }, function (error) {
-        defer.reject(error);
-      });
+      var pagedTotalData = new PagedCollection($scope.config.customModule, null, null);
+      pagedTotalData.loadByPost(filters).then(function () {
+        if (pagedTotalData.fieldRows.length === 0) {
 
-      return defer.promise;
+          return;
+        }
+        var data = pagedTotalData.fieldRows[0][$scope.config.customModuleField].value;
+        if (!data) {
+          data = {};
+        } else {
+          $scope.processing = false;
+          chartData = data['data'];
+          createDataToPlot(chartData);
+        }
+      })
     }
 
+    //render text if no data found or nodes and links are not formed by the fetched data
```

Comment:
The function parameter is missing here. errorMessage used on line no. 237.

Reply:
errorMessage is saved and updated in global variable